commit 7950d023c562823345892aac2e7c6a49f8de9ad1
Author: Artem Bityutskiy <Artem.Bityutskiy@nokia.com>
Date:   Fri Nov 19 17:05:36 2010 +0200

    UBI: use raw mtd read function in debugging code
    
    This change affects only the debugging code. Namely, use mtd->read()
    function instead of ubi_io_read() to avoid bit-flips injection
    (ubi_dbg_is_bitflip()) which we do not want on the debugging path.
    
    Signed-off-by: Artem Bityutskiy <Artem.Bityutskiy@nokia.com>

diff --git a/drivers/mtd/ubi/io.c b/drivers/mtd/ubi/io.c
index 339a74f11c0b..6f90a6530149 100644
--- a/drivers/mtd/ubi/io.c
+++ b/drivers/mtd/ubi/io.c
@@ -1325,10 +1325,12 @@ int ubi_dbg_check_write(struct ubi_device *ubi, const void *buf, int pnum,
 			int offset, int len)
 {
 	int err, i;
+	size_t read;
+	loff_t addr = (loff_t)pnum * ubi->peb_size + offset;
 
 	mutex_lock(&ubi->dbg_buf_mutex);
-	err = ubi_io_read(ubi, ubi->dbg_peb_buf, pnum, offset, len);
-	if (err)
+	err = ubi->mtd->read(ubi->mtd, addr, len, &read, ubi->dbg_peb_buf);
+	if (err && err != -EUCLEAN)
 		goto out_unlock;
 
 	for (i = 0; i < len; i++) {